### PR TITLE
Gauss law for magnetic field

### PR DESCRIPTION
**Pull request by Zulusssss**
add condition

### CODE REVIEW

File: symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from sympy import Eq
```
Comment (blackyblack) [inline symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py:1]: `divergence_of_magnetic_induction_field_is_zero`

File: symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+from sympy.core import Equality
+from symplyphysics import (units, Symbol)
+from symplyphysics.core.fields.operators import divergence_operator
+from symplyphysics.core.fields.vector_field import VectorField
+
+## Description
+## The divergence of magnetic induction at some point is equal to zero. A simpler formulation of the law
```
Comment (blackyblack) [inline symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py:8]: "induction at some point"
At what point?

File: symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+## is that magnetic charges not exist.
+
+## Law is: div(B) = 0, where
+## B - magnetic induction (vector field),
+## div - divergence (the sum of partial derivatives in coordinates).
+
+time = Symbol('time', units.time)
```
Comment (blackyblack) [inline symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py:15]: Unused.

File: symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+def gauss_law_for_magnetic_field(magnetic_induction_: VectorField) -> Equality:
+    divergence_magnetic_induction = divergence_operator(magnetic_induction_)
+    law = Eq(divergence_magnetic_induction, 0)
+    return law
```
Comment (blackyblack) [inline symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py:20]: Rename function to condition, i.e. `magnetic_field_divergence_condition` and make it shorter:
`return Eq(divergence_magnetic_induction, 0)`

File: symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+## The divergence of magnetic induction at some point is equal to zero. A simpler formulation of the law
+## is that magnetic charges not exist.
+
+## Law is: div(B) = 0, where
+## B - magnetic induction (vector field),
+## div - divergence (the sum of partial derivatives in coordinates).
+
+time = Symbol('time', units.time)
+
+def gauss_law_for_magnetic_field(magnetic_induction_: VectorField) -> Equality:
```
Comment (blackyblack) [inline symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py:17]: Add test for this condition.

Comment (Zulusssss) [reply symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py:17]: > Add test for this condition.

What should I test? I can set a field whose divergence will be 0 and set it, but it probably doesn't make sense.

Comment (blackyblack) [reply symplyphysics/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_0.py:17]: It perfectly makes sense. Do it.

File: test/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_zero_test.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    magnetic_amplitude = Quantity(1 * prefixes.kilo * units.tesla)
+    magnetic_induction = VectorField(lambda point: [(point.x / Quantity(1 * units.meter)) * magnetic_amplitude,
+                                                      (-point.y / Quantity(1 * units.meter)) * magnetic_amplitude,
+                                                      magnetic_amplitude])
+    return Args(magnetic_induction=magnetic_induction)
+
+
+def test_basic_magnetic_field_divergence_condition(test_args: Args) -> None:
+    result = divergence_cond.magnetic_field_divergence_condition(test_args.magnetic_induction)
+    assert_equal(result.args[0], 0 * units.tesla / units.meter)
```
Comment (blackyblack) [inline test/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_zero_test.py:29]: Mypy does not like it. Probably you have to convert left side to some number.

Comment (blackyblack) [inline test/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_zero_test.py:29]: I think there should be some function to verify, that equality is true. Try to find one.
Or use `eq.lhs == eq.rhs`

File: test/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_zero_test.py [hunk 7]
```diff
@@ -26,4 +27,4 @@ def test_args_fixture() -> Args:
 
 def test_basic_magnetic_field_divergence_condition(test_args: Args) -> None:
     result = divergence_cond.magnetic_field_divergence_condition(test_args.magnetic_induction)
-    assert_equal(result.args[0], 0 * units.tesla / units.meter)
+    assert simplify(result.lhs) == result.rhs
```
Comment (blackyblack) [inline test/conditions/electricity/maxwell_equations/divergence_of_magnetic_induction_field_is_zero_test.py:30]: Also add negative test, when equation is not zero.

**Review by blackyblack — APPROVED** [review]
